src/control/lib/netdetect/netdetect_test.go

```diff
@@ -122,3 +122,79 @@ func TestValidateNetworkConfig(t *testing.T) {
 		AssertEqual(t, err, nil, "Network device configuration is invalid - NUMA node does not match")
 	}
 }
+
+// TestDeviceAlias uses XML topology data to simulate real systems.
+// hwloc will use this topology for queries instead of the local system running the test.
+// This test verifies that GetDeviceAlias() is capable of performing this lookup
+// This particular test verifies the expected lookup path that uses a device found
+// on the system device list, and expected to find the related sibling.
+func TestDeviceAlias(t *testing.T) {
+	tests := []struct {
```

Comment:
It would be nicer to organize these tests with the following format:
```go
for name, tc := range map[string]struct{
  // struct def
}{
  "test name 1": {
    // test parameters
  },
  "test name 2": {
    // test parameters
 },
}{
  t.Run(name, func(t *testing.T) {
    // test body
  })
}
```
I have been recommending this layout for table-driven tests for a few reasons:

1. Naming each test case clearly makes it easy to understand which testcase failed
2. Using subtests improves test output by individually displaying each subtest
3. Using a map of structs injects some randomness into test order execution as map iteration is nondeterministic

Another thing to consider is using types in your test configuration struct that have useful zero values (e.g. bool, pointer-to-whatever). If a given subtest is expected to return an error, then you can define an error that you're expecting to see. Otherwise, the expected error is nil and you don't need to explicitly define that.

See for example these tests: https://github.com/daos-stack/daos/blob/master/src/control/common/env_utils_test.go

Reply:
Ok, I will rework it.

Reply:
I really like the new change in the test format.  The ability to clearly show what test failed is great.